sunpy/image/resample.py

```diff
@@ -82,31 +82,44 @@ def resample(orig, dimensions, method='linear', center=False, minusone=False):
 def _resample_nearest_linear(orig, dimensions, method, offset, m1):
     """
     Resample Map using either linear or nearest interpolation.
+
+    Parameters
+    ----------
+    orig :
+        Original data.
+    dimensions :
+        Dimensions of resamled data.
+    offset :
+        Either 0 or 0.5, depending on whether interpolation is at the edge or
+        centers of bins.
+    m1 :
+        For ``orig.shape = (i,j)`` & new dimensions ``= (x,y)``, if set to `False`
+        (default) ``orig`` is resampled by factors of ``(i/x) * (j/y)``,
+        otherwise ``orig`` is resampled by ``(i-1)/(x-1) * (j-1)/(y-1)``.
+        This prevents extrapolation one element beyond bounds of input array.
```

Comment:
I know this sentence is just copied from the docstring of the user-facing function, but I'm confused regardless.  What does this sentence mean?  What does it have to do with a parameter that just modifies the scaling factor?

Reply:
It's gibberish to me too, and seems to have been lifted directy from the IDL CONGRID docs: https://www.harrisgeospatial.com/docs/CONGRID.html

Reply:
Ah, I get it now.  It actually makes more sense with the flipped order of sentences in the IDL documentation.

Consider `i=3` and `x=6`.

With `m1=0` (`minusone=False` in the user-facing function):
```
orig: |         |         |         |
new : |    |    |    |    |    |    |
```
3 pixels are being mapped to 6 pixels, so the factor is `i/x=3/6=0.5`.

The issue is that the outermost pixels of the output array require extrapolating off of the edges of the input array.  If you instead resample by the "minus one" dimensions, the outermost pixels in the input and output arrays are lined up, so no extrapolation is needed.

With `m1=1` (`minusone=True` in the user-facing function):
```
orig: |         |         |         |
new :    |   |   |   |   |   |   |
```
That is, here `2=0.5+1+0.5` pixels are being resampled to `5=0.5+4+0.5` pixels, which is why the factor is , so the factor is `(i-1)/(x-1)=2/5=0.4`.